### ISSUE TITLE
how to custom metric, such as kappa

### ISSUE THREAD

**Issue by hust-kevin** [pytorch/ignite#551]

**Comment by vfdev-5** [comment]
@hust-kevin could you provide a detailed definition of this metric ?

**Comment by hust-kevin** [comment]
@vfdev-5  I want to metric the Quadratic Weighted Kappa on the validation set,  sklearn has method [sklearn.metrics.cohen_kappa_score](https://docs.w3cub.com/scikit_learn/modules/generated/sklearn.metrics.cohen_kappa_score/) to compute it, ,but i don't kown how to use it in ignite
but

**Comment by vfdev-5** [comment]
@hust-kevin there are two ways to do this. If in your case it is possible to store whole history of `y1` and `y2` (as defined in sklearn `sklearn.metrics.cohen_kappa_score(y1, y2, labels=None, weights=None, sample_weight=None)`), then you can use [`EpochMetric`](https://pytorch.org/ignite/metrics.html#ignite.metrics.EpochMetric) with defined `compute_fn` as in the case of [`AveragePrecision`](https://pytorch.org/ignite/_modules/ignite/contrib/metrics/average_precision.html#AveragePrecision).


HTH

**Comment by hust-kevin** [comment]
thanks, I solved it !

**Comment by Vietdung113** [comment]
> thanks, I solved it !

can u push your solution?
i still don't know how to do that?
